### PR TITLE
Create new product for SUSE Linux Enterprise Micro

### PR DESCRIPTION
**Pull request by jlausuch**
After the rename form SUSE MicroOS to SLE Micro, it
makes sense to do some adjustments to differenciate
between OpenSUSE MicroOS and this one.

- Related ticket: https://progress.opensuse.org/issues/87880
- VRs: [DVD](http://fromm.arch.suse.de/tests/91) [RAW](http://fromm.arch.suse.de/tests/92) [RT](http://fromm.arch.suse.de/tests/90) [Containers](http://fromm.arch.suse.de/tests/94)

### CODE REVIEW

File: products/sle-micro/main.pm [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+my $distri = testapi::get_required_var('CASEDIR') . '/lib/susedistribution.pm';
+require $distri;
+testapi::set_distribution(susedistribution->new());
+
+#######################
+# Testing starts here #
+#######################
+return 1 if load_yaml_schedule;
+
+
```
Comment (b10n1k) [inline products/sle-micro/main.pm:23]: extra line

File: lib/version_utils.pm [hunk 2]
```diff
@@ -230,14 +227,22 @@ sub is_microos {
...
         return check_version($filter, $version, qr/\d{1,}\.\d/);
     }
     else {
         return $flavor eq $filter;    # Specific FLAVOR selector
     }
 }
 
+=head2 is_sle_micro
+Check if distribution is SUSE Linux Enterprise Micro
+=cut
```
Comment (b10n1k) [inline lib/version_utils.pm:240]: nitpicking but i believe the perlpod requires a space after  the `=head2` and maybe before the `=end` too. you can check the output http://podwebview.odyniec.net/

File: lib/version_utils.pm [hunk 3]
```diff
@@ -200,9 +201,9 @@ sub check_version {
 }
 
 =head2 is_microos
-Check if distribution is SUSE MicroOS or openSUSE MicroOS with optional filter:
+Check if distribution is openSUSE MicroOS with optional filter:
```
Comment (b10n1k) [inline lib/version_utils.pm:204]: Could you verify the output here  http://podwebview.odyniec.net/ please?

**Review by b10n1k — CHANGES_REQUESTED** [review]
Very neat and complete. i had some really minor findings only if you could take care of them.

File: products/sle-micro/main.pm (hunk 1 above)
Comment (jlausuch) [reply products/sle-micro/main.pm]: done

File: lib/version_utils.pm (hunk 2 above)
Comment (jlausuch) [reply lib/version_utils.pm:242]: done

File: lib/version_utils.pm (hunk 3 above)
Comment (jlausuch) [reply lib/version_utils.pm:205]: done

**Review by mloviska — APPROVED** [review]
LGTM, just the checks :)

**Comment by DimStar77** [comment]
Can you please make sure to clone jobhs from group/1 to the DEV group when you want to do tests/development?

I understand you need to do development, that is appreciated; and looking that things don't break for TW is even more appreciated.

But if you keep on cloning jobs in group/1 you keepon blocking Tumbleweed snapshots, as the totest manager needs a clean state to release the product

There is a development tumbleweed group intentionally to work on stuff that does not yet work:
  https://openqa.opensuse.org/group_overview/38

**Comment by jlausuch** [comment]
> Can you please make sure to clone jobs from group/1 to the DEV group when you want to do tests/development?
> 
> I understand you need to do development, that is appreciated; and looking that things don't break for TW is even more appreciated.
> 
> But if you keep on cloning jobs in group/1 you keepon blocking Tumbleweed snapshots, as the totest manager needs a clean state to release the product
> 
> There is a development tumbleweed group intentionally to work on stuff that does not yet work:
> https://openqa.opensuse.org/group_overview/38

Yeah, I realized after doing it and I'm sorry. I'm cloning the jobs locally after adapting my openqa instance. Thank you.

**Comment by Vogtinator** [comment]
> > Can you please make sure to clone jobs from group/1 to the DEV group when you want to do tests/development?
> > I understand you need to do development, that is appreciated; and looking that things don't break for TW is even more appreciated.
> > But if you keep on cloning jobs in group/1 you keepon blocking Tumbleweed snapshots, as the totest manager needs a clean state to release the product
> > There is a development tumbleweed group intentionally to work on stuff that does not yet work:
> > https://openqa.opensuse.org/group_overview/38
> 
> Yeah, I realized after doing it and I'm sorry. I'm cloning the jobs locally after adapting my openqa instance. Thank you.

You can use o3 for that just fine, just make sure to pass `_GROUP=38` to `openqa-clone-job` or use `openqa-clone-custom-git-refspec`, which does that automatically.

File: lib/version_utils.pm [hunk 4]
```diff
@@ -213,10 +215,6 @@ sub is_microos {
     return 0 unless $distri && $distri =~ /microos/;
     return 1 unless $filter;
 
-    # SUSE MicroOS will have versions 5.X, 6.X., the rest is openSUSE
-    my $type = $version =~ /^[56]\./ ? 'suse' : 'opensuse';
-    return $filter eq $type if ($filter =~ /opensuse|^suse/);
-
     my $version_is_tw = ($version =~ /Tumbleweed/ || $version =~ /^Staging:/);
```
Comment (asmorodskyi) [inline lib/version_utils.pm:218]: currently this function will be used only for o3 , right ? do we have Staging in o3 ? maybe we can simplify it even more ?

Comment (jlausuch) [reply lib/version_utils.pm:218]: Maybe, but I wanted to leave it as it was originally. If it needs improvement, I would leave it for another PR.

File: lib/version_utils.pm [hunk 5]
```diff
@@ -230,14 +228,23 @@ sub is_microos {
...
         return $flavor eq $filter;    # Specific FLAVOR selector
     }
 }
 
+=head2 is_sle_micro
+
+Check if distribution is SUSE Linux Enterprise Micro
+=cut
+sub is_sle_micro {
+    return 1 if check_var('DISTRI', 'sle-micro');
```
Comment (asmorodskyi) [inline lib/version_utils.pm:244]:
```suggestion
    return check_var('DISTRI', 'sle-micro');
```

and you can drop one line below

**Review by asmorodskyi — APPROVED** [review]
LGTM

**Comment by okurz** [comment]
s/differenciate/differentiate/

btw, if you triggered certain openQA tests by mistake you can either update them, e.g. move to another group using the API or also delete them completely, e.g. using `openqa-cli … -X delete jobs/<nr>`

File: lib/version_utils.pm (hunk 5 above)
Comment (jlausuch) [reply lib/version_utils.pm]: done

**Review by b10n1k — APPROVED** [review]
LGTM

**Comment by jlausuch** [comment]
Latest VRs:
[microos-Tumbleweed-MicroOS-Image-x86_64-Build20210118-microos@64bit](http://fromm.arch.suse.de/tests/96)
[microos-Tumbleweed-DVD-x86_64-Build20210118-microos@64bit](http://fromm.arch.suse.de/tests/95)

[sle-micro_update@64bit](http://fromm.arch.suse.de/tests/103)
[sle-micro_installation+update@64bit](http://fromm.arch.suse.de/tests/100)
[sle-micro_autoyast@64bit](http://fromm.arch.suse.de/tests/102)
[sle-micro_installation+update@uefi](http://fromm.arch.suse.de/tests/101)
[sle-micro_containers@64bit](http://fromm.arch.suse.de/tests/109)
[sle-micro_update@64bit](http://fromm.arch.suse.de/tests/108)

**Comment by DimStar77** [comment]
apropos:

> between OpenSUSE MicroOS and this one.

Can we manage to get at least SUSE employees to spell 'openSUSE' correctly? :)

**Pull request merged**